api/v1alpha1/spaceprovisionerconfig_types.go

```diff
@@ -50,10 +53,28 @@ type SpaceProvisionerCapacityThresholds struct {
 	MaxMemoryUtilizationPercent uint `json:"maxMemoryUtilizationPercent,omitempty"`
 }
 
+// ConsumedCapacity describes the capacity of the cluster consumed by the spaces
+// currently provisioned to it.
+type ConsumedCapacity struct {
+	// MemoryUsagePercentPerNodeRole is the percent of the memory used per node role (eg. worker, master)
+	MemoryUsagePercentPerNodeRole map[string]int `json:"memoryUsagePercentPerNode"`
```

Comment:
I might be wrong but it should require also these markers
```suggestion
	// MemoryUsagePercentPerNodeRole is the percent of the memory used per node role (eg. worker, master)
	// +optional
        // +mapType=atomic
	MemoryUsagePercentPerNodeRole map[string]int `json:"memoryUsagePercentPerNode,omitempty"`
```

Reply:
I will add the `+maptype=atomic`, thanks for pointing it out. As for optional and omitempty, I don't these should be added in this case, see my reply to your comment on the `SpaceCount`. That said, I also noticed the discrepancy between the field name in go and json, so I'll fix that, too :)